onecodex/viz/_bargraph.py

```diff
@@ -0,0 +1,124 @@
+            Analysis will be restricted to abundances of taxa at the specified level.
+        normalize : 'auto' or `bool`, optional
+            Convert read counts to relative abundances such that each sample sums to 1.0. Setting
+            'auto' will choose automatically based on the data.
+        return_chart : `bool`, optional
+            When True, return an `altair.Chart` object instead of displaying the resulting plot in
+            the current notebook.
+        top_n : `int`, optional
+            Display the top N most abundant taxa in the entire cohort of samples.
+        threshold : `float`
+            Display only taxa that are more abundant that this threshold in one or more samples.
+        title : `string`, optional
+            Text label at the top of the plot.
+        xlabel : `string`, optional
+            Text label along the horizontal axis.
+        ylabel : `string`, optional
+            Text label along the vertical axis.
+        tooltip : `string` or `list`, optional
+            A string or list containing strings representing metadata fields. When a point in the
+            plot is hovered over, the value of the metadata associated with that sample will be
+            displayed in a modal.
+    """
+
+    def plot_bargraph(
+        self,
+        rank="auto",
+        normalize="auto",
+        threshold="auto",
+        top_n="auto",
+        title=None,
+        xlabel=None,
+        ylabel=None,
+        tooltip=None,
+        return_chart=False,
+    ):
+
+        if rank is None:
+            raise OneCodexException(
+                "Please specify a rank or 'auto' to choose automatically"
+            )
+
+        if not (threshold or top_n):
+            raise OneCodexException("Please specify at least one of: threshold, top_n")
+
+        if top_n == "auto" and threshold == "auto":
+            top_n = 10
+            threshold = None
+        elif top_n == "auto" and threshold != "auto":
+            top_n = None
+        elif top_n != "auto" and threshold == "auto":
+            threshold = None
+
+        df = self.to_df(
+            rank=rank,
+            normalize=normalize,
+            top_n=top_n,
+            threshold=threshold,
+            table_format="long",
+        )
+
+        if tooltip:
+            if not isinstance(tooltip, list):
+                tooltip = [tooltip]
+        else:
+            tooltip = []
+
+        # takes metadata columns and returns a dataframe with just those columns
+        # renames columns in the case where columns are taxids
+        magic_metadata, magic_fields = self._metadata_fetch(tooltip)
+
+        for f in tooltip:
+            df[magic_fields[f]] = magic_metadata[magic_fields[f]][
+                df["classification_id"]
+            ].tolist()
+
+        # add sample filenames
+        df["display_name"] = self.metadata["_display_name"][
+            df["classification_id"]
+        ].tolist()
```

Comment:
`_metadata_fetch` has a secret keyword where the string "Label" will be converted to the contents of `_display_name`. The benefit here is that the x axis title will then be "Label" if `xlabel=None`. Maybe add "Label" to `tooltip` and get rid of this?

Reply:
set default `xaxis="Sample"`